### DISCUSSION TITLE
Solution Version: incrementing solution version while exporting

### DISCUSSION THREAD

**Discussion by anshupandey0407** [microsoft/powerplatform-actions#280] [Q&A] [answered]
Hello Team,

I would like to increase the solution version number while exporting so that the new version is used for importing in target environment. is there any action available for this task?

please help

**Comment by robcaretta-thrivent** [comment]
Interested in this as well.

**Comment by paulbreuler** [comment, marked as answer]
If I understand the ask correctly then my suggestion would be to do this as part of you pipeline using `pac solution version --strategy Solution `.

Example: 

```
egrep "<Version>.*</Version>" src/Other/Solution.xml
<Version>1.0.1</Version>
```

```
pac solution version --strategy Solution
Build version updated for I:\Work-Repos\test\src\Other\Solution.xml.
```

```
egrep "<Version>.*</Version>" src/Other/Solution.xml
    <Version>1.0.2</Version>
```

**The equivalent in GitHub Actions for the Power Platform is update-solution-version**
https://github.com/microsoft/powerplatform-actions/blob/main/update-solution-version/action.yml

**Comment by robcaretta-thrivent** [comment]
@paulbreuler - How might this be accomplished entirely in GitHub actions? The use case would be the same, but I'd like to automatically get and increment the version. I'm working from the [actions lab examples](https://github.com/microsoft/powerplatform-actions-lab). 

Regarding that update-solution-version link it references an option to utilize a "tracker-file". Where might we find an example of that format? And is that manually managed or automatic?

**Reply by paulbreuler** [reply]
I am working on some updates for tracker file and versioning in general. So, my TL;DR is wait for the February release. 

Regarding how to get the tracker file it is initially generated by the tooling. 

All versioning solutions are automated, however, in the case that you are updating files there must be some persistence. For Git Tags you can simply track tags on the repository, which is great because you do not need to make changes to your source code. With methods requiring changes to files, those files must be committed to the repository to leverage updates in the next CI/CD cycle. 

I'll setup up an example in another repo as I'm testing things out and endeavor to make them templates/composite actions/reusable workflows. That way you can benefit from my work and see an example setup. 

https://github.com/paulbreuler/powerplatform-cli-workflow-templates

**Reply by pratham-p** [reply]
@paulbreuler Do you have any reference docs or some example on how to use "filetracking" strategy? We are looking to see how can we track changes to the files in a solution and automatically bump up the version if there are changes detected.

**Comment by fergalflat** [comment]
Any update to this?
